extensions/standard-processors/processors/SplitJson.cpp

```diff
@@ -0,0 +1,133 @@
+
+#include "core/ProcessSession.h"
+#include "core/ProcessContext.h"
+#include "core/Resource.h"
+#include "utils/ProcessorConfigUtils.h"
+#include "utils/Id.h"
+
+#include "jsoncons_ext/jsonpath/jsonpath.hpp"
+
+namespace org::apache::nifi::minifi::processors {
+
+void SplitJson::initialize() {
+  setSupportedProperties(Properties);
+  setSupportedRelationships(Relationships);
+}
+
+void SplitJson::onSchedule(core::ProcessContext& context, core::ProcessSessionFactory&) {
+  json_path_expression_ = utils::parseProperty(context, SplitJson::JsonPathExpression);
+  null_value_representation_ = utils::parseEnumProperty<split_json::NullValueRepresentationOption>(context, SplitJson::NullValueRepresentation);
+}
+
+std::optional<jsoncons::json> SplitJson::queryArrayUsingJsonPath(core::ProcessSession& session, const std::shared_ptr<core::FlowFile>& flow_file) const {
+  const auto json_string = to_string(session.readBuffer(flow_file));
+  if (json_string.empty()) {
+    logger_->log_error("FlowFile content is empty, transferring to Failure relationship");
+    return std::nullopt;
+  }
+
+  jsoncons::json json_object;
+  try {
+    json_object = jsoncons::json::parse(json_string);
+  } catch (const jsoncons::json_exception& e) {
+    logger_->log_error("FlowFile content is not a valid JSON document, transferring to Failure relationship: {}", e.what());
+    return std::nullopt;
+  }
+
+  jsoncons::json query_result;
+  try {
+    query_result = jsoncons::jsonpath::json_query(json_object, json_path_expression_);
+  } catch (const jsoncons::jsonpath::jsonpath_error& e) {
+    logger_->log_error("Invalid JSON path expression '{}' set in the 'JsonPath Expression' property: {}", json_path_expression_, e.what());
+    return std::nullopt;
+  }
+
+  return query_result;
+}
+
+std::string SplitJson::jsonValueToString(const jsoncons::json& json_value) const {
+  if (json_value.is_null()) {
+    return null_value_representation_ == split_json::NullValueRepresentationOption::EmptyString ? "" : "null";
+  }
+
+  if (json_value.is_string()) {
+    return json_value.as<std::string>();
+  }
+
+  return json_value.to_string();
+}
+
+void SplitJson::onTrigger(core::ProcessContext& context, core::ProcessSession& session) {
+  auto flow_file = session.get();
+  if (!flow_file) {
+    context.yield();
+    return;
+  }
+
+  auto query_result = queryArrayUsingJsonPath(session, flow_file);
+  if (!query_result) {
+    session.transfer(flow_file, Failure);
+    return;
+  }
+
+  gsl_Assert(query_result->is_array());
+  if (query_result->empty()) {
+    logger_->log_error("JSON Path expression '{}' did not match the input flow file content, transferring to Failure relationship", json_path_expression_);
+    session.transfer(flow_file, Failure);
+    return;
+  }
+
+  if (query_result->size() == 1 && !query_result.value()[0].is_array()) {
+    logger_->log_error("JSON Path expression '{}' did not return an array, transferring to Failure relationship", json_path_expression_);
```

Comment:
```suggestion
    logger_->log_error("JSON Path expression '{}' did not return an array, transferring to the 'failure' relationship", json_path_expression_);
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1980/commits/0aa867f69a964c3d3f0a1562750bd81218fc6434